sklearn/utils/tests/test_extmath.py

```diff
@@ -159,29 +170,6 @@ def check_randomized_svd_low_rank(dtype):
             np.dot(U[:, :k], Vt[:k, :]), np.dot(Ua, Va), decimal=decimal
         )
 
-        # check the sparse matrix representation
-        X = sparse.csr_matrix(X)
-
-        # compute the singular values of X using the fast approximate method
-        Ua, sa, Va = randomized_svd(
-            X, k, power_iteration_normalizer=normalizer, random_state=0
-        )
-        if dtype.kind == "f":
-            assert Ua.dtype == dtype
-            assert sa.dtype == dtype
-            assert Va.dtype == dtype
-        else:
-            assert Ua.dtype.kind == "f"
-            assert sa.dtype.kind == "f"
-            assert Va.dtype.kind == "f"
-
-        assert_almost_equal(s[:rank], sa[:rank], decimal=decimal)
```

Comment:
I would not change the test here (or I would parametrize with random seed). We checked `:rank` and now `:k` that is different.

Reply:
Basically we could still program it that we have a sparse matrix we check for `:rank` while `:k` for dense.